jgrapht-io/src/main/java/org/jgrapht/io/DOTExporter.java

```diff
@@ -262,7 +262,11 @@ private void renderAttributes(PrintWriter out, String labelName, Map<String, Att
             }
         }
         if (labelName != null) {
-            out.print("label=\"" + labelName + "\" ");
+            if (DOTUtils.isQuoted(labelName)) {
```

Comment:
This doesn't seem right to me.  If a label provider produces something containing double quotes (regardless of whether they are at the beginning and end), we should be escaping them according to the spec.  So the correct fix would be to apply escaping uniformly.

Reply:
Maybe you are right, I have added this because quoted id is valid identifier but generated diagram is corrupted. I think this is for ids like *"two words id"*. You can't declare such id without quoting and they shouldn't be quoted two times.

https://github.com/mariuszs/jgrapht/blob/9be45689add961ee60c37a19e1252cb8223ec9b5/jgrapht-io/src/main/java/org/jgrapht/io/DOTUtils.java#L56

Reply:
> An ID is one of the following:
> any double-quoted string ("...") possibly containing escaped quotes (\")1;

And yes, quotes declared inside should be quoted.

Reply:
So, according to the specification surrounding quoting should be removed.

Reply:
Ok, and the reason for this is because quoting in diagram is not required.

Fragment of valid diagram:

    A2 -> A3 [label=g];
    B2 -> B3 [label="g'"];

but in our generator we are always adding quoting.

Reply:
We cannot escape id like `"foo"`, because according do spec escaping is possible only for string like this `\"foo\"`

```
In quoted strings in DOT, the only escaped character is double-quote ("). That is,
in quoted strings, the dyad \" is converted to "; all other characters are left 
unchanged. In particular, \\ remains \\. Layout engines may apply additional escape 
sequences.
```

Reply:
I see what you mean.  But this is quirky in that normally an interface such as VertexProvider shouldn't be taking care of the quoting internally; that should be left to the exporter based on the text of the identifier.  In other words, instead of failing when isValidID is false, just automatically quote it (escaping as necessary).

Here's another reason to consider.  The spec says that if a DOT keyword is used as an identifier, it needs to be quoted.  Now, imagine you're using an arbitrary VertexProvider implementation, and it returns a string such as 'strict', which just happens to be a DOT keyword.  Now when you put this VertexProvider together with DOTExporter, you produce invalid results.  Instead, the DOTExporter should auto-quote keywords.

Reply:
Hmm, I'm not sure. It is safe for us to quote everything, because this way we cannot generate label with `strict` like in this invalid example:
   
``` 
 graph graphname {
  a [ label = "strict"]; //valid
  b [ label = strict]; //invalid
 }
```
so quoting everything is ok. But I don't get why quoting already quoted identifiers is wrong. 

For example:

```
 graph foo {
  a [ label = "\"foobar\""]; //valid
  b [ label = ""foobar""]; //invalid 
 }
```
Label surrounded with double quoted and no escaped value is always wrong.

I will leave this case to you :)

Reply:
Oh, you mean we should escape double quoted charactes inside this label?

Reply:
Right.